eZ/Publish/Core/Persistence/Legacy/Content/FieldValue/Converter/ImageConverter.php

```diff
@@ -176,6 +179,32 @@ public function toFieldValue(StorageFieldValue $value, FieldValue $fieldValue)
         $fieldValue->data = $this->parseLegacyXml($value->dataText);
     }
 
+    public function toStorageFieldDefinition(FieldDefinition $fieldDef, StorageFieldDefinition $storageDef): void
+    {
+        $validators = $fieldDef->fieldTypeConstraints->validators;
+
+        $storageDef->dataInt1 = $validators['FileSizeValidator']['maxFileSize'] ?? 0;
+        $storageDef->dataInt2 = (int)($validators['AlternativeTextValidator']['required'] ?? 0);
+    }
+
+    public function toFieldDefinition(StorageFieldDefinition $storageDef, FieldDefinition $fieldDef): void
+    {
+        $fieldDef->fieldTypeConstraints = new FieldTypeConstraints(
+            [
+                'validators' => [
+                    'FileSizeValidator' => [
+                        'maxFileSize' => ($storageDef->dataInt1 != 0
```

Comment:
What about strong comparison here?
```suggestion
                        'maxFileSize' => ($storageDef->dataInt1 !== 0
```
Not sure if it's only me but those parentheses are adding this feeling that there is something more here, but they seems redundant?

Reply:
If that was supposed to be a "fix" for a float comparison then `abs($storageDef->dataInt1) < PHP_FLOAT_EPSILON` should be used instead?

Reply:
I am almost sure that `maxFileSize` is int not float.
And you cant store floats in `dataIntX` properties, right? We have `dataFloatX` for that.

Reply:
> I am almost sure that `maxFileSize` is int not float.
> And you cant store floats in `dataIntX` properties, right? We have `dataFloatX` for that.

`dataIntX` should never contain anything other than int or null.